lib/strategies/webapp-strategy.js

```diff
@@ -406,7 +406,8 @@ function generateAuthorizationUrl(options, req, strategy) {
 
 function retrieveTokens(formData, strategy) {
```

Comment:
This function flow is still complicated with callacks and nested promises, but looks better than the previous version so we're in the right way 👍